src/utils/ucc_status.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "ucc/api/ucc_status.h"
+#include <stdio.h>
+
+const char *ucc_status_string(ucc_status_t status)
+{
+    static char error_str[128] = {0};
+
+    switch (status) {
+    case UCC_OK:
+        return "Success";
+    case UCC_INPROGRESS:
+        return "Operation is posted and is in progress";
+    case UCC_OPERATION_INITIALIZED:
+        return "Operation initialized but not posted";
+    case UCC_ERR_NOT_SUPPORTED:
+        return "Operation is not supported";
+    case UCC_ERR_NOT_IMPLEMENTED:
+        return "Function is not implemented";
```

Comment:
not necessarily function: can be just some flag support. it would call it just "not implemented"